### PR TITLE
divide flux(1) help output into sections

### PR DESCRIPTION
**Pull request by grondo**
This PR reworks the output of `flux help` to use sections (like `git help`) to be a little more user friendly.
Also, some "plumbing" commands (like `flux kvs`, `flux get/set/lsattr`) are dropped so that all the commands listed are the most important commands for new users.

Also added a little note at the end of `flux help` output to use `flux help COMMAND` to get help on a specific command.

The method of building the `flux help` output from tags in man page source is kept for now, though I'm unsure if that is the correct long term approach. However, this should be a pretty big step forward for now.

New output:
```
Usage: flux [OPTIONS] COMMAND ARGS
  -h, --help             Display this message.
  -v, --verbose          Be verbose about environment and command search
  -V, --version          Display command and component versions
  -p, --parent           Set environment of parent instead of current instance

For general Flux documentation, please visit
    https://flux-framework.readthedocs.io

run and submit jobs, allocate resources
   submit             submit a job to a Flux instance
   run                run a Flux job interactively
   bulksubmit         submit jobs in bulk to a Flux instance
   alloc              allocate a new Flux instance for interactive use
   batch              submit a batch script to Flux

list and interact with jobs
   jobs               list jobs submitted to Flux
   top                display running Flux jobs
   pstree             display job hierarchies
   pgrep/pkill        search or cancel matching jobs
   job                cancel jobs, get job status, etc (see: flux help job) 
   proxy              proxy connections to Flux jobs and instances

get resource, queue and other instance information
   resource           list/manipulate Flux resource status
   queue              list and manipulate flux queues
   overlay            Show flux overlay network status
   uptime             Tell how long Flux has been up and running

other useful commands
   start              bootstrap a local Flux instance
   version            Display flux version information
   config             Manage/query Flux configuration
   env                Print the flux environment or execute a command inside it

See 'flux help COMMAND' for more information about a specific command.
```

### CODE REVIEW

File: doc/manpages.py
Comment (garlick) [inline doc/manpages.py:20]: Could we maybe add a note here that the order affects `flux help` output?

Someone might decide to alphabetize the list or something.

**Review by garlick — APPROVED** [review]
Nice improvement!  One minor comment inline.

File: doc/manpages.py
Comment (grondo) [reply doc/manpages.py]: Good suggestion. I went ahead and did that.

**Comment by vsoch** [comment]
omg! I don't know how I missed this - it's pure beauty!! So much better what I had suggested before! :heart_eyes:

**Review by vsoch — APPROVED** [review]
I can't review the C++ code, but if it works and produces that flux help output and tests pass, I so strongly approve!!

**Comment by grondo** [comment]
Great, thanks @vsoch! Was just waiting for your approval and now we'll go ahead and merge this one!

**Pull request merged**
